lambda/js/src/test/scala/feral/lambda/ContextPlatformSuite.scala

```diff
@@ -0,0 +1,227 @@
+      override def logStreamName: String = generatedLogStreamName
+      override def identity: js.UndefOr[facade.CognitoIdentity] = generatedIdentity
+      override def clientContext: js.UndefOr[facade.ClientContext] = generatedClientContext
+      override def getRemainingTimeInMillis(): Double = generatedRemainingTimeInMillis
+
+      override def toString(): String =
+        s"Context(functionName=$functionName, functionVersion=$functionVersion, invokedFunctionArn=$invokedFunctionArn, memoryLimitInMB=$memoryLimitInMB, awsRequestId=$awsRequestId, logGroupName=$logGroupName, logStreamName=$logStreamName, identity=$identity, clientContext=$clientContext, remainingTimeInMillis=${getRemainingTimeInMillis()}"
+    }
+  }
+
+  private implicit def optionCompare[A, B](
+      implicit C: Compare[A, B]): Compare[Option[A], Option[B]] =
+    new Compare[Option[A], Option[B]] {
+      override def isEqual(obtained: Option[A], expected: Option[B]): Boolean =
+        (obtained, expected).mapN(C.isEqual).getOrElse(obtained.isEmpty && expected.isEmpty)
+    }
+
+  private implicit val compareCognitoIdentity
+      : Compare[CognitoIdentity, facade.CognitoIdentity] =
+    new Compare[CognitoIdentity, facade.CognitoIdentity] {
+      override def isEqual(
+          obtained: CognitoIdentity,
+          expected: facade.CognitoIdentity): Boolean =
+        obtained.identityId == expected.cognitoIdentityId && obtained.identityPoolId == expected.cognitoIdentityPoolId
+    }
+
+  private implicit val compareClientContextClient
+      : Compare[ClientContextClient, facade.ClientContextClient] =
+    new Compare[ClientContextClient, facade.ClientContextClient] {
+      override def isEqual(
+          obtained: ClientContextClient,
+          expected: facade.ClientContextClient): Boolean =
+        obtained.installationId == expected.installationId &&
+          obtained.appTitle == expected.appTitle &&
+          obtained.appVersionName == expected.appVersionName &&
+          obtained.appVersionCode == expected.appVersionCode &&
+          obtained.appPackageName == expected.appPackageName
+    }
+
+  private implicit val compareClientContextEnv
+      : Compare[ClientContextEnv, facade.ClientContextEnv] =
+    new Compare[ClientContextEnv, facade.ClientContextEnv] {
+      override def isEqual(
+          obtained: ClientContextEnv,
+          expected: facade.ClientContextEnv): Boolean =
+        obtained.platformVersion == expected.platformVersion &&
+          obtained.platform == expected.platform &&
+          obtained.make == expected.make &&
+          obtained.model == expected.model &&
+          obtained.locale == expected.locale
+    }
+
+  private implicit val compareJsonObjectWithUndefOrAny
+      : Compare[JsonObject, js.UndefOr[js.Any]] = new Compare[JsonObject, js.UndefOr[js.Any]] {
+    override def isEqual(obtained: JsonObject, expected: js.UndefOr[js.Any]): Boolean =
+      expected
+        .toOption
+        .map(js.JSON.stringify(_))
```

Comment:
Instead of going through a JSON string and parsing, Circe can directly decode a JS object.